.github/workflows/nightly.yml

```diff
@@ -0,0 +1,104 @@
+name: Nightly
+
+on:
+  push:
+  ## Allow manual invocation
+  #workflow_dispatch:
+  #schedule:
+  #  - cron: '0 5 * * *' # UTC
+
+env:
+ DIST_ROOT: '/ipns/dist.ipfs.io' # content root used for calculating diff to build
+ GO_IPFS_VER: 'v0.9.1'           # go-ipfs daemon used for chunking and applying diff
+ CLUSTER_CTL_VER: 'v0.14.0'      # ipfs-cluster-ctl used for pinning
+
+concurrency:
+  group: nightly
+  cancel-in-progress: true
+
+jobs:
+  prepare-matrix:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - id: set-matrix
+        run: echo "::set-output name=matrix::$(jq -nc '$ARGS.positional' --args $(ls ./dists -1))"
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+
+  build:
+    runs-on: "ubuntu-latest"
+    needs: prepare-matrix
+    strategy:
+      max-parallel: 15 # avoid using all workers from the org pool
+      fail-fast: false
+      matrix:
+        dist_name: ${{ fromJson(needs.prepare-matrix.outputs.matrix) }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup IPFS
+        run: ./scripts/ci/setup-ipfs.sh
+        env:
+          CLUSTER_USER: ${{ secrets.CLUSTER_USER }}
+          CLUSTER_PASSWORD: ${{ secrets.CLUSTER_PASSWORD }}
+        timeout-minutes: 5
+      - run: cd ./dists/${{ matrix.dist_name }} && make nightly
+      - run: ./dockerized make publish
+      - name: Create issue if build failed
+        uses: actions/github-script@v4
```

Comment:
:information_source: Third-party turn-key solutions exists, but I decided to give `actions/github-script` a try.  It is an official action provided by GitHub, provides JS porcelain for running arbitrary API calls. 

Debugging is a bit painful, but being written in JS could make it more dev friendly than bash+curl, allowing more people to maintain/extend.  `actions/github-script` is extremely powerful, only caveat is that it needs a [personal access token](https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token) to do anything outside the current repo.

We may leverage it even more for developer flow automation tailored to our needs. For example, it could automatically create/label issues  and add them to specific column in specific project.